frontend/public/components/utils/file-input.tsx

```diff
@@ -28,13 +37,20 @@ export class FileInput extends React.Component<FileInputProps, FileInputState> {
       });
       return;
     }
+    let fileIsBinary = false;
     const reader = new FileReader();
     reader.onload = () => {
-      const input = reader.result;
-      this.props.onChange({
-        fileData: input,
-        fileName: file.name,
-      });
+      const input = reader.result as string;
+      if (containsNonPrintableCharacters(input) && !fileIsBinary) {
+        fileIsBinary = true;
+        reader.readAsBinaryString(file);
+      } else {
+        this.props.onChange({
+          fileData: fileIsBinary ? Base64.encode(input) : input,
```

Comment:
I wouldn't Base64 encode here. For secrets, the value is always encoded whether it's binary or not. I'd let the caller encode the value.

Reply:
@spadgett ok, removed base64 encoding here.